src/librustc_mir/error_codes.rs

```diff
@@ -2424,6 +2424,36 @@ const fn foo() -> impl T {
 ```
 "##,
 
+E0729: r##"
+Support for Non-Lexical Lifetimes (NLL) has been included in the Rust compiler
+since 1.31, and has been enabled on the 2015 edition since 1.36. The new borrow
+checker for NLL uncovered some bugs in the old borrow checker, which in some
+cases allowed unsound code to compile, resulting in memory safety issues.
+
+### What do I do?
+
+Change your code so the warning does no longer trigger. For backwards
+compatibility, this unsound code may still compile right now. However, at some
+point in the future, the compiler will no longer accept this code and will
+throw a hard error.
+
+### Shouldn't you fix the old borrow checker?
+
+The old borrow checker has known soundness issues that are basically impossible
+to fix. The new NLL-based borrow checker is the fix.
+
+### Can I turn these warnings into errors by denying a lint?
+
+No.
+
+### When are these warnings going to turn into errors?
+
+FIXME
+
+### Why do I get this message with code that doesn't involve borrowing?
+
+There are some known bugs that trigger this message.
```

Comment:
what is this a reference to? I'm not claiming that you're wrong, but I want to understand what we're talking about.

Reply:
#60450

Reply:
okay. The phrasing "There are some known bugs ..." is a bit ambiguous, since the phrase "known bugs" there could denote:

 1. bugs in `rustc` itself that are known to the rustc developers (and presumably would be fixed at some point in the future)
 2. buggy coding patterns in user code, where the patterns are known and documented
 3. coding patterns that may or may not be buggy, where we are using github issues (thus "bugs") to document such cases, and where `rustc` itself may be revised in the future to change its behavior when it encounters such patterns.

I actually do not object to the ambiguity here, since I suspect all three cases might arise at various points, so I'm okay with letting all such cases be covered by the phrasing here, at least for now.